app/dot/cpp/repository/repository/BaseRepository.java

```diff
@@ -118,9 +158,28 @@ public T getFirstSorted(Sort sort) {
     }
   }
 
+  public void saveWithHistory(T entity) {
+    final var currentEntity = findById(entity.getId());
+
+    entity.setModifiedAt(Instant.now().getEpochSecond());
+
+    if (currentEntity == null) {
+      // entity creation
+      entity.setTrackingId(UUID.randomUUID().toString());
+      entity.setModifiedComment(INITIAL);
+    } else {
+      // set null id in order to generate a new, unique one in the history collection
+      currentEntity.setId(null);
```

Comment:
Is this the only way to do it?

Reply:
Why would I do this differently if Mongo generates an id automatically when the id is not set? Before this, we also let Mongo generate our ids.